src/routes/todo.routes/createToDo.ts

```diff
@@ -7,22 +7,41 @@ import { saveTodo } from "../../repository/todo.repository/todoRepository";
 import { Api400Error } from "../../services/error.services/api400Error";
 import { Api500Error } from "../../services/error.services/api500Error";
 import { httpStatusCodes } from "../../services/error.services/httpStatusCodes";
+import { ToDo } from "../../repository/todo.repository/types";
 
 export function createToDoRouter(): Router {
   const todoRouter = Router();
 
-  return todoRouter.post("/", (req: Request, res: Response) => {
-    const inputValidationOutcome = validateBody(req.body);
-    if (inputValidationOutcome.isErr()) {
-      throw new Api400Error(inputValidationOutcome.error);
-    }
-    const { title, description } = req.body;
-    const toDo = createToDo(title, description);
+  return todoRouter.post(
+    "/",
+    (req: Request, res: Response, next: NextFunction) => {
+      let todo = {} as ToDo;
+      try {
+        validateBody(req.body).match(
+          (value) => {
+            console.log(value);
+          },
+          (e) => {
+            throw new Api400Error(e);
+          }
+        );
 
-    const saveOutcome = saveTodo(toDo);
-    if (saveOutcome.isErr()) {
-      throw new Api500Error(saveOutcome.error);
+        const { title, description } = req.body;
+        createToDo(title, description).match(
+          (value) => {
+            todo = value;
+            saveTodo(todo);
+          },
+          (e) => {
+            throw new Api500Error(e);
```

Comment:
We shouldn't throw 500 errors on purpose. We should define what could possibly go wrong when creating a ToDo and handle all those edge-cases separately.